Backend/DynamicQR/DynamicQR.Infrastructure/Services/QrCodeTargetRepositoryService.cs

```diff
@@ -12,53 +12,68 @@ public sealed class QrCodeTargetRepositoryService : IQrCodeTargetRepositoryServi
 
-        return response.IsError;
+        if (response.IsError)
+            throw new StorageException(response.ReasonPhrase);
     }
 
     public async Task<QrCodeTarget> ReadAsync(string id, CancellationToken cancellationToken)
     {
         ArgumentNullException.ThrowIfNull(id);
 
-        var data = await _tableClient.GetEntityIfExistsAsync<Entities.QrCodeTarget>("qrCodeTarget", id, cancellationToken: cancellationToken);
+        Azure.NullableResponse<Entities.QrCodeTarget> data = await _tableClient.GetEntityIfExistsAsync<Entities.QrCodeTarget>("Value", id, cancellationToken: cancellationToken);
 
         if (data.HasValue)
-        {
-            return new QrCodeTarget
-            {
-                QrCodeId = data.Value.QrCodeId,
-                Value = data.Value.Value
-            };
-        }
+            return data.Value.ToCore();
 
-        throw new EntryPointNotFoundException();
+        throw new StorageException();
     }
 
-    public async Task<bool> UpdateAsync(QrCodeTarget qrCodeTarget, CancellationToken cancellationToken)
+    public async Task UpdateAsync(QrCodeTarget qrCodeTarget, CancellationToken cancellationToken)
     {
         ArgumentNullException.ThrowIfNull(qrCodeTarget);
 
-        //var data = await _tableClient.GetEntityIfExistsAsync<Entities.QrCodeTarget>("qrCodeTarget", qrCodeTarget.QrCodeId, cancellationToken: cancellationToken);
-        //if (!data.HasValue)
-        throw new EntryPointNotFoundException();
+        QrCodeTarget qrCodeTargetFound = await ReadAsync(qrCodeTarget.QrCodeId, cancellationToken);
+
+        if (qrCodeTargetFound != null)
+        {
+            qrCodeTargetFound.Value = qrCodeTarget.Value;
+
+            var data = QrCodeTargetMappers.ToInfrastructure(qrCodeTargetFound);
+
+            Azure.Response response = await _tableClient.UpdateEntityAsync(data, Azure.ETag.All, TableUpdateMode.Merge, cancellationToken);
 
-        //await _tableClient.UpdateEntityAsync(, poem.Value.ETag);
+            if (response.IsError)
+                throw new StorageException(response.ReasonPhrase);
+        }
+        else
+            throw new StorageException();
     }
 
-    public async Task<bool> DeleteAsync(string id, CancellationToken cancellationToken)
+    public async Task DeleteAsync(string id, CancellationToken cancellationToken)
     {
         ArgumentNullException.ThrowIfNull(id);
 
-        throw new NotImplementedException();
+        QrCodeTarget qrCodeTargetFound = await ReadAsync(id, cancellationToken);
+
+        if (qrCodeTargetFound != null)
+        {
+            Azure.Response response = await _tableClient.DeleteEntityAsync(qrCodeTargetFound.QrCodeId, qrCodeTargetFound.QrCodeId, Azure.ETag.All, cancellationToken);
+
+            if (response.IsError)
+                throw new StorageException(response.ReasonPhrase);
+        }
+        else
+            throw new StorageException();
```

Comment:
Dit zou dan ook een ItemNotFoundException (of specifieker QrCodeTargetNotFoundException) moeten worden